misc-utils/waitpid.c

```diff
@@ -118,17 +161,22 @@ static int parse_options(int argc, char **argv)
 {
 	int c;
 	static const struct option longopts[] = {
-		{ "verbose", no_argument, NULL, 'v' },
-		{ "version", no_argument, NULL, 'V' },
-		{ "help",    no_argument, NULL, 'h' },
+		{ "verbose", no_argument,       NULL, 'v' },
+		{ "timeout", required_argument, NULL, 't' },
+		{ "version", no_argument,       NULL, 'V' },
+		{ "help",    no_argument,       NULL, 'h' },
 		{ NULL, 0, NULL, 0 },
 	};
 
-	while ((c = getopt_long (argc, argv, "vVh", longopts, NULL)) != -1) {
+	while ((c = getopt_long (argc, argv, "vVht:", longopts, NULL)) != -1) {
 		switch (c) {
 		case 'v':
 			verbose = true;
 			break;
+		case 't':
+			timeout = strtou64_or_err(
+					optarg, _("Could not parse timeout"));
```

Comment:
For example, in flock or irqtop we use `strtotimeval_or_err()`, which allows specifying the timeout as a decimal number (1.5). Maybe we can use it also for waitpid, as timerfd is based on sec+nsec. (It would be nice to support it also for the kill command, etc.)

Reply:
Done, I added a new `strtotimespec_or_err()` as timerfds are using `struct timespec`.
As for kill, that is currently using milliseconds as CLI argument, while the functions are parsing seconds.